build/Dockerfile.dev

```diff
@@ -0,0 +1,17 @@
+FROM registry.access.redhat.com/ubi9/go-toolset:1.19 as build
+USER 0
+RUN mkdir /build
+WORKDIR /build
+# cache dependencies
+ADD go.mod go.sum /build/
+RUN go mod download
+COPY . .
+RUN make build strip GO=go
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+COPY --from=build /build/pbapi /pbapi
+COPY --from=build /build/pbworker /pbworker
+COPY --from=build /build/pbstatuser /pbstatuser
+COPY --from=build /build/pbmigrate /pbmigrate
+USER 1001
+CMD ["/pbapi"]
```

Comment:
I think for development purposes, you do not need to use ubi-minimal at all. In fact, it is counter productive as you get rid of the Go compiler cache which is one of the reasons why Go compiler is very fast. Also you can take advantage of `go run` command. Something like (pseudo code):

```
FROM registry.access.redhat.com/ubi9/go-toolset:1.19 as build
USER 0
RUN mkdir /build
WORKDIR /build
COPY . .
RUN make prep GO=go
CMD ["go", "run", "/build/cmd/pbapi/main.go"]
```

This way you can easily just add new and new commits on top of this single container, everytime you do a change the subsequent "go run" will be fast as files will be already pre-compiled. Also you can change the entrypoint easily for worker or statuser.

Reply:
> I think for development purposes, you do not need to use ubi-minimal at all. 

Makes sense 👍  a tiny tweak i've added is using [CompileDaemon](github.com/githubnemo/CompileDaemon) for deamomzing go and allowing live reloading when .*go files have change, saves tone of time during dev.

Reply:
Not a big fan of that as our binary is 70MB and compilation is relatively slow but I am fine with this :-)